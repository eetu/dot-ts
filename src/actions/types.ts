export const DOT_ADD = 'DOT_ADD';
export const LANG_EN = 'LANG_EN';
export const LANG_FI = 'LANG_FI';
