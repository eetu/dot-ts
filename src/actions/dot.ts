import * as types from './types';

export type DotActions = {
  [types.DOT_ADD]: {
    type: typeof types.DOT_ADD,
  },
};

export type DotAction = DotActions[keyof DotActions];

export const actions = {
  add: (): DotActions[typeof types.DOT_ADD] => ({
    type: types.DOT_ADD,
  }),
};
