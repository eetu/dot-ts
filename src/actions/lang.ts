import * as types from './types';

export type LangActions = {
  [types.LANG_FI]: {
    type: typeof types.LANG_FI;
  };
  [types.LANG_EN]: {
    type: typeof types.LANG_EN;
  };
};

export type LangAction = LangActions[keyof LangActions];

export const actions = {
  langFi: (): LangActions[typeof types.LANG_FI] => ({
    type: types.LANG_FI,
  }),
  langEn: (): LangActions[typeof types.LANG_EN] => ({
    type: types.LANG_EN,
  }),
};
