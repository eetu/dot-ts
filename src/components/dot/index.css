.Dot {
  font-weight: 600;
}
