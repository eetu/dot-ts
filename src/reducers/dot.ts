import * as types from '../actions/types';
import { DotAction } from '../actions/dot';

export type DotState = {
  dots: string;
};

const initialState = {
  dots: ''
};

const dot = (state = initialState, action: DotAction) => {
  switch (action.type) {
    case types.DOT_ADD:
      return Object.assign({}, state, {
        dots: state.dots + '.'
      });
    default:
      return state;
  }
};

export default dot;
