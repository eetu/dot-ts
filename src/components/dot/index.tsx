import * as React from 'react';
import { DotState } from '../../reducers/dot';
import { actions } from '../../actions/dot';
import { TranslationFunction } from 'react-i18next';

import './index.css';

interface DotComponentProps {
  state: DotState;
  actions: typeof actions;
  t: TranslationFunction;
}

const Dot: React.StatelessComponent<DotComponentProps> = (
  props: DotComponentProps,
) => (
  <div>
    <div>
      {props.t('Dots')}: <span className="Dot">{props.state.dots}</span>
    </div>
    <button onClick={props.actions.add}>{props.t('Add')}</button>
  </div>
);

export default Dot;
