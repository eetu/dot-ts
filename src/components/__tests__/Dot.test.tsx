import * as React from 'react';
import Dot from '../dot';
import { actions } from '../../actions/dot';
import { create } from 'react-test-renderer';

test('Dot should render', () => {
  const state = { dots: '' };

  const component = create(
    <Dot actions={actions} state={state}/>
  );

  expect(component).toMatchSnapshot();
});
