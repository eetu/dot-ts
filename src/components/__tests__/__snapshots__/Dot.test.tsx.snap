// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot should render 1`] = `
<div>
  <div>
    dots: 
    <span
      className="Dot"
    >
      
    </span>
  </div>
  <button
    onClick={[Function]}
  >
    Add dots
  </button>
</div>
`;
